### DISCUSSION TITLE
Homebrew Private Repositories

### DISCUSSION THREAD

**Discussion by ekristen** [renovatebot/renovate#13504] [Request Help]
Can renovate support private repositories with homebrew? I've setup hostRules but it looks like the file stream to hash the archive file doesn't include the token?

**Comment by rarkins** [comment]
I think you're referring to a couple of `http.stream` calls like this: https://github.com/renovatebot/renovate/blob/db211c594c61df5670ff8a2fad422a58c5dd4f4a/lib/manager/homebrew/update.ts#L165-L167

I wasn't aware that this call bypassed our usual auth from hostRules, but here it is: https://github.com/renovatebot/renovate/blob/db211c594c61df5670ff8a2fad422a58c5dd4f4a/lib/util/http/index.ts#L287-L303

Here is where auth is normally applied: https://github.com/renovatebot/renovate/blob/db211c594c61df5670ff8a2fad422a58c5dd4f4a/lib/util/http/index.ts#L150

So I guess this might be about `http.stream()` in our util folder not using hostRules rather than about Homebrew specifically. Some of these might also be affected if so:

![image](https://user-images.githubusercontent.com/6311784/149510939-65e7d1e4-f605-4922-91ef-0764e2ed568a.png)

**Reply by ekristen** [reply]
I dug into this a bit more last night, I couldn't confirm if it was bypassing or not, but I don't think it is. Unfortunately private repo download URLs have to be access via the API vs the browser urls which are sort of hard coded into the code to look for them at `https://github.com/owner/repo/....` so I think the module would need an overhaul to support private repos with homebrew.
